internal/gengapic/example_test.go

```diff
@@ -276,6 +277,92 @@ func TestExample(t *testing.T) {
 	}
 }
 
+func TestGenSnippetFile(t *testing.T) {
+	var g generator
+	g.imports = map[pbinfo.ImportSpec]bool{}
+	g.serviceConfig = &serviceconfig.Service{
+		Apis: []*apipb.Api{
+			{Name: "google.cloud.bigquery.migration.v2.MigrationService"},
+		},
+	}
+
+	protoPkg := "google.cloud.bigquery.migration.v2"
+	libPkg := "cloud.google.com/go/bigquery/migration/apiv2"
+	pkgName := "bigquerymigration"
+	var err error
+	g.snippetMetadata, err = snippets.NewMetadata(protoPkg, libPkg, pkgName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	inputType := &descriptor.DescriptorProto{
+		Name: proto.String("CreateMigrationWorkflowRequest"),
+	}
+	outputType := &descriptor.DescriptorProto{
+		Name: proto.String("MigrationWorkflow"),
+	}
+
+	file := &descriptor.FileDescriptorProto{
+		Options: &descriptor.FileOptions{
+			GoPackage: proto.String("cloud.google.com/go/bigquery/migration/apiv2/migrationpb"),
+		},
+		Package: proto.String(protoPkg),
+	}
+
+	files := []*descriptor.FileDescriptorProto{}
+	g.descInfo = pbinfo.Of(files)
+	for _, typ := range []*descriptor.DescriptorProto{
+		inputType, outputType,
```

Comment:
nit: Please always provide struct keys.

Reply:
`inputType` and `outputType` hold pointers to `descriptor.DescriptorProto` instances. In this line they are being added to to a new slice of `[]*descriptor.DescriptorProto`. That's ok, isn't it?

Reply:
I can change this to 4 lines of code that directly call `g.descInfo` methods for `inputType` and `outputType`. This loop pattern is well established in the tests. It makes more sense when there are more elements in the slice, for example [here](https://github.com/googleapis/gapic-generator-go/blob/d5ed011a154765d2ff8fa555ccaad35ce4fd1dcd/internal/gengapic/example_test.go#L121-L126). But if you ask me, loops almost always reduce the clarity of the tests. I personally think it's usually better to have more lines of simple setup statements. I was just sticking to established style here.